src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Pattern.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeRefactorings.ConvertIfToSwitch;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertIfToSwitch
+{
+    partial class CSharpConvertIfToSwitchCodeRefactoringProvider
+    {
+        public abstract class Pattern
```

Comment:
It can be private but it can't be in VB. Weird. 

Pattern doesn't have anything in common for VB and C# and we need concrete instance of each therefore it is passed as a type parameter. I had an idea that involves contravariant override but it's currently not available. I could probably simulate it though. I'll see what can I do when I try ISyntaxFacts that you've suggested below. As a related note, I also tried to use SyntaxGenerator but it seems it's not updated since pattern matching so that didn't help.